tuf/api/metadata.py

```diff
@@ -1296,3 +1296,33 @@ def to_dict(self) -> Dict[str, Any]:
     def update(self, fileinfo: TargetFile) -> None:
         """Assigns passed target file info to meta dict."""
         self.targets[fileinfo.path] = fileinfo
+
+    def add_key(self, role: str, key: Key) -> None:
+        """Adds new signing key for delegated role 'role'.
+
+        Raises:
+            ValueError: If there are no delegations or delegate role 'role'.
+        """
+        if self.delegations is None or role not in self.delegations.roles:
+            raise ValueError("Delegations doesn't exist")
```

Comment:
This is not the correct error message.

Also I still think the docstrings should not talk about implementation details (whether delegations is none or not) here or in remove_key(): they should talk about what the callers mistake was -- but this is a nitpick.